lib/ethpy/ethpy/test_fixtures/deploy_hyperdrive.py

```diff
@@ -0,0 +1,242 @@
+        ],
+    )
+
+    return base_token_contract, factory_contract
+
+
+def deploy_and_initialize_hyperdrive(
+    web3: Web3,
+    base_token_contract: Contract,
+    factory_contract: Contract,
+    deploy_account: LocalAccount,
+) -> str:
+    """Calls the hyperdrive factory to deploy and initialize new hyperdrive contract
+
+    Arguments
+    --------
+    web3 : Web3
+        web3 provider object
+    base_token_contract : Contract
+        The base token contract
+    factory_contract : Contract
+        The hyperdrive factory contract
+    deploy_account: LocalAccount
+        The account that deploys the contracts
+
+    Returns
+    -------
+    str
+        The deployed hyperdrive contract address
+    """
+    # TODO parameterize these parameters
+    # pylint: disable=too-many-locals
+    # Initial hyperdrive settings
+    initial_contribution = int(100_000_000e18)
+    initial_share_price = int(1e18)
+    minimum_share_reserves = int(10e18)
+    position_duration = 604800  # 1 week
+    checkpoint_duration = 3600  # 1 hour
+    time_stretch = _calculateTimeStretch(int(0.05e18))
+    oracle_size = 10
+    update_gap = 3600  # 1 hour
+    initial_fixed_rate = int(0.05e18)
+
+    deploy_addr = Web3.to_checksum_address(deploy_account.address)
+
+    # Mint base tokens
+    # Need to pass signature instead of function name since multiple mint functions
+    tx_receipt = smart_contract_transact(
+        web3, base_token_contract, deploy_account, "mint(address,uint256)", deploy_addr, initial_contribution
+    )
+    tx_receipt = smart_contract_transact(
+        web3, base_token_contract, deploy_account, "approve", factory_contract.address, initial_contribution
+    )
+
+    # Call factory to make hyperdrive market
+    # Some of these pool info configurations don't do anything, as the factory is overwriting them
+    pool_config = (
+        base_token_contract.address,
+        initial_share_price,
+        minimum_share_reserves,
+        position_duration,
+        checkpoint_duration,
+        time_stretch,
+        deploy_addr,  # governance, overwritten by factory
+        deploy_addr,  # feeCollector, overwritten by factory
+        (0, 0, 0),  # fees, overwritten by factory
+        oracle_size,  # oracleSize
+        update_gap,
+    )
+    tx_receipt = smart_contract_transact(
+        web3,
+        factory_contract,
+        deploy_account,
+        "deployAndInitialize",
+        # Function arguments
+        pool_config,
+        [],  # new bytes[](0)
+        initial_contribution,
+        initial_fixed_rate,  # fixedRate
+    )
+
+    logs = get_transaction_logs(factory_contract, tx_receipt)
+    hyperdrive_address = None
+    for log in logs:
+        if log["event"] == "GovernanceUpdated":
+            hyperdrive_address = log["address"]
+    if hyperdrive_address is None:
+        raise AssertionError("Generating hyperdrive contract didn't return address")
+
+    # TODO do I return the contract or address here?
```

Comment:
Do you mean the hyperdrive contract? I don't see that in here. The address is probably fine.